hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,576 @@
+		// }
+		// else
+		// {
+		// ROS_ERROR_STREAM("Failed to connect to '"<< moveit_run_path_client_.getService()<<"' service");
+		// exit(-1);
+		// }
+
+		ROS_INFO_STREAM("Task '"<<__FUNCTION__<<"' completed");
+		// ROS_INFO_STREAM("Descartes and MoveIt! are happy.");
+	}
+
+	void HybridPlanner::update()
+	{
+		moveit_group_ptr_->getCurrentState();
+		kinematic_model_ = kinematic_state_->getRobotModel();
+		joint_model_group_ = kinematic_model_->getJointModelGroup(config_.group_name); 
+		ROS_INFO_STREAM("MoveIt! members updated");
+		//TODO show the status and values of member variables ROS_INFO_STREAM
+	}
+
+	void HybridPlanner::loadParameters()
+	{
+		ros::NodeHandle ph("~");
+		ros::NodeHandle nh;
+		//TOCHECK if namespaces are needed in the node handle constructors 
+		//TOCHECK required parameters
+		//TODO MoveIt params are hardcoded
+		if(ph.getParam("group_name",config_.group_name) &&
+		  ph.getParam("tip_link",config_.tip_link) &&
+		  ph.getParam("base_link",config_.base_link) &&
+		  ph.getParam("world_frame",config_.world_frame) &&
+		  ph.getParam("trajectory/time_delay",config_.time_delay) &&
+		  ph.getParam("trajectory/seed_pose",config_.seed_pose) &&
+		  ph.getParam("visualization/min_point_distance",config_.min_point_distance) &&
+		  nh.getParam("controller_joint_names",config_.joint_names) )
+		{
+		ROS_INFO_STREAM("Loaded Descartes parameters");
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to load Descartes parameters");
+		exit(-1);
+		}
+
+		//TOCHECK Set the following hereor should be loaded seperately for each free space segment?
+		moveit_group_ptr_->setPlanningTime(config_.plan_time);
+		moveit_group_ptr_->allowReplanning(config_.moveit_replan);
+		ROS_INFO_STREAM("Loaded MoveIt! parameters");//TOCHECK add negative debug statement, if needed with if(){} block. 
+
+		ROS_INFO_STREAM("Task '"<<__FUNCTION__<<"' completed");
+	}
+
+
+	bool HybridPlanner::appendDescartesCartPoint(Eigen::Affine3d& pose)
+	{	
+		//TOCHECK pass const parameter? 
+		//TODO allocate size of descartes_traj_input_ or just suffice by push_back()?
+	  	using namespace descartes_core;
+  		using namespace descartes_trajectory;
+	  	TrajectoryPtPtr pt = TrajectoryPtPtr(new CartTrajectoryPt(TolerancedFrame(pose)));
+	  	
+	  	if(!pt)
+  		{	
+  			descartes_traj_input_.push_back(pt);
+	  	// publishing trajectory pose for visualization
+	  	// publishPosesMarkers(pose);
+  			return 1;	
+  		}
+  
+	 	else
+	 	{
+	  		ROS_WARN("Failed for append point. Moving on anyway.");
+	  		return 0;
+	  	}	
+	}
+
+	bool HybridPlanner::appendTolerancedDescartesCartPoint(Eigen::Affine3d& pose)
+	{
+		using namespace descartes_core;
+	  	using namespace descartes_trajectory;
+	  	TrajectoryPtPtr pt = TrajectoryPtPtr(new AxialSymmetricPt(pose, ORIENTATION_INCREMENT, 
+	  															AxialSymmetricPt::Z_AXIS));
+	  	descartes_traj_input_.push_back(pt);
+	}
+
+	bool HybridPlanner::appendDescartesCartPointSegment(EigenSTL::vector_Affine3d poses)
```

Comment:
Yes. `TOCHECK` flag here https://github.com/madratman/hamilton/blob/indigo-devel/hamilton/include/hamilton/hybrid_planner.h#L124
Initially, I wanted them to be void. But bool seemed to be a good option for debugging. But I didn't implement it yet.